Algorithm.CSharp/NumeraiSignalExportDemonstrationAlgorithm.cs

```diff
@@ -31,184 +31,91 @@ namespace QuantConnect.Algorithm.CSharp
+            Schedule.On(DateRules.EveryDay(_etfSymbol), TimeRules.At(13, 0, TimeZones.Utc), SubmitSignals);
 
-            // Set Numerai signal export provider
-            SignalExport.AddSignalExportProviders(new NumeraiSignalExport(_numeraiPublicId, _numeraiSecretId, _numeraiModelId, _numeraiFilename));
+            // Add the Numerai signal export provider
+            // Numerai Public ID: This value is provided by Numerai Signals in their main webpage once you've logged in
+            // and created a API key. See (https://signals.numer.ai/account)
+            var numeraiPublicId = "";
 
-            SetWarmUp(100);
+            // Numerai Public ID: This value is provided by Numerai Signals in their main webpage once you've logged in
+            // and created a API key. See (https://signals.numer.ai/account)
+            var numeraiSecretId = "";
+
+            // Numerai Model ID: This value is provided by Numerai Signals in their main webpage once you've logged in
+            // and created a model. See (https://signals.numer.ai/models)
+            var numeraiModelId = "";
+
+            var numeraiFilename = ""; // (Optional) Replace this value with your submission filename 
+            SignalExport.AddSignalExportProviders(new NumeraiSignalExport(numeraiPublicId, numeraiSecretId, numeraiModelId, numeraiFilename));
         }
 
-        /// <summary>
-        /// Reduce the quantity of holdings for SPY or increase it, depending the case, 
-        /// when the EMA's indicators crosses between themselves, then send a signal to
-        /// Numerai API
-        /// </summary>
-        /// <param name="slice"></param>
-        public override void OnData(Slice slice)
+        public void SubmitSignals()
         {
-            if (IsWarmingUp) return;
+            // Select the subset of ETF constituents we can trade
+            var symbols = _securities.Where(security => security.HasData).Select(security => security.Symbol);
+            if (symbols.Count() == 0)
+            {
+                return;
+            }
+
+            // Get historical data
+            // var history = History(symbols, 22, Resolution.Daily);
 
-            // Place an order as soon as possible to send a signal.
-            if (_firstCall)
+            // Create portfolio targets
+            var denominator = symbols.Count() * (symbols.Count() + 1) / 2.0m; // sum of 1, 2, ..., symbols.Count()
+            symbols = symbols.OrderBy(symbol => symbol);
+            var targets = new List<PortfolioTarget>();
+            for (var i = 0; i < symbols.Count(); i++)
             {
-                SetHoldings("SPY", 0.1);
-                _targets[0] = new PortfolioTarget(Portfolio["SPY"].Symbol, (decimal)0.1);
-                SignalExport.SetTargetPortfolio(_targets);
-                _firstCall = false;
+                targets.Add(new PortfolioTarget(symbols.ToList()[i], (i + 1) / denominator));
```

Comment:
`symbols` is an enumerable, every operation `.Count()/.ToList()/.OrderBy` is re enumerating the linq statement. Suggest moving `OrderBy` at the top same to the `ToList`

I'm a bit confused by `var denominator = symbols.Count() * (symbols.Count() + 1) / 2.0m;` & `(i + 1) / denominator)`, for example for `10` symbols this will be 1/55, 2/55 until reaching 10? maybe just simpler to do `1 / count` for all of them?
Similar comments for python algorithm 👍

Reply:
We can't give an equal weight to each security because Numerai requires at least one of them to be unique.
In this example, we give a linear allocation to each asset. This is correct:
> for 10 symbols this will be 1/55, 2/55 until reaching 10?